packages/atclient/src/atclient.c

```diff
@@ -369,6 +387,9 @@ exit: {
   free(root_cmd);
   free(from_cmd);
   free(pkam_cmd);
+  if (should_free_atserver_host) {
+    free(atserver_host);
+  }
```

Comment:
free the atserver_host if it was allocated during automatic lookup (wasn't provided by options)